test/jdk/tools/jar/InputFilesTest.java

```diff
@@ -153,6 +153,51 @@ public void test6() throws IOException {
         jar("cf test.jar --release 9 -C test1 a -C test2 a");
     }
 
+    /**
+     * Containing non-existent file in the file list
+     * The final jar should not be created and correct error message should be caught.
+     * IOException is triggered as expected.
+     */
+    @Test(expectedExceptions = {IOException.class})
+    public void testNonExistentFileInput() throws IOException {
+        touch("existingTestFile.txt");
+        onCompletion = () -> rm("existingTestFile.txt");
+        try {
+            jar("cf test.jar existingTestFile.txt nonExistentTestFile.txt");
+        } catch (IOException e) {
+            Assert.assertEquals(e.getMessage().trim(), "nonExistentTestFile.txt : no such file or directory");
+            Assert.assertTrue(Files.notExists(Path.of("test.jar")), "Jar file should not be created.");
+            throw e;
+        }
```

Comment:
There should be a `fail("jar tool unexpectedly completed successfully")` here so that we assert that the `jar` call actually fails. Same comment for the other new test method.